tests/mapped_struct.py

```diff
@@ -2075,14 +2075,64 @@ class ProxiedDictPackingTest(unittest.TestCase, CollectionPackingTestHelpers, Di
         {'a': 'a2', 'b': 'b2', 'c': 'c2'},
         {1: 10, 2: 20, 3: 30},
         {'a': frozenset(), 'b': (1, 2), 'c': 1.0, 'd': [1, 2], 'e': dict(a=1) },
-        {0: 42, 'a1_@!': 69, 3.5: 'uhhhh', (1, 2, 3): "four-five-six"},
+        {0: 42, 'a1_@!': 69, -3.5: 'uhhhh', (1, 2, 3): "four-five-six"},
         {frozenset([1, 2]) : 97.9},
         {1.0: "test floats equivalent to integers"},
         {None: 3},
         {(1,2,3): 4, (-1,3): 7},
         {frozenset([1,2,3]): 4, frozenset([-1,3]): 7},
+        {float("inf"): 0, float("-inf"): 1, -0.0001: 2, -123456.: 3}
     ]
 
+class StableHashTest(unittest.TestCase):
+
+    def assertHashesOK(self, values):
```

Comment:
All the tests that use `assertHashesOk` fail to check the stability of the hash values. `assertHashesOk` should get the hash-value pairs instead.

Reply:
There are 2 types of tests in the class `StableHashTest`:

The first kind is the one that receives the value array, computes the hash values once, then computes them again, and checks that they are equal (to ensure hash values don't depend on any global state) and that they are of type `int` or `long`.

The second kind uses the same value array, but with the addition of the precomputed hashes in order to test for stability. They are the ones with the suffix `Fixed`.

Reply:
If you think about it, the second test is stricter than the first, and implies correctness under the first tests' conditions, so you don't need the first kind of tests, just the second kind.

Reply:
But you have different test values, and the test values for the "fixed" kind is a smaller set. So better unify them. Merge all the test values into a single "Fixed" test for each kind of object and that's both less code and more thorough testing.